daemon/oci_opts.go

```diff
@@ -0,0 +1,21 @@
+package daemon
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/containers"
+	coci "github.com/containerd/containerd/oci"
+	"github.com/docker/docker/container"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// WithConsoleSize sets the initial console size
+func WithConsoleSize(c *container.Container) coci.SpecOpts {
+	return func(ctx context.Context, _ coci.Client, _ *containers.Container, s *coci.Spec) error {
+		s.Process.ConsoleSize = &specs.Box{
+			Height: c.HostConfig.ConsoleSize[0],
+			Width:  c.HostConfig.ConsoleSize[1],
+		}
```

Comment:
To be on the safe side, perhaps we should add a check here, so that we don't override the default (`nil`) value if there's no size to be set;

```suggestion
		if c.HostConfig.ConsoleSize[0] > 0 || c.HostConfig.ConsoleSize[1] > 0 {
			s.Process.ConsoleSize = &specs.Box{
				Height: c.HostConfig.ConsoleSize[0],
				Width:  c.HostConfig.ConsoleSize[1],
			}
		}
```

Reply:
FWIW; we should start looking at using the containerd options for these things, but they use a different signature (all options in our local package basically take a full Config, so that requires quote some refactoring); e.g. in this case, we'd have to replace with https://github.com/moby/moby/blob/4cac624fefefb7bb6640f205587d0dc0a9651b27/vendor/github.com/containerd/containerd/oci/spec_opts.go#L246-L258


But that's definitely out of scope for this PR